src/replica/split/replica_split_manager.cpp

```diff
@@ -645,9 +644,242 @@ void replica_split_manager::parent_check_sync_point_commit(decree sync_point) //
     }
 }
 
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica_split_manager::update_child_group_partition_count(
+    int new_partition_count) // on primary parent
+{
+    if (status() != partition_status::PS_PRIMARY || _split_status != split_status::SPLITTING) {
+        derror_replica(
+            "wrong partition status or wrong split status, partition_status={}, split_status={}",
+            enum_to_string(status()),
+            enum_to_string(_split_status));
+        _stub->split_replica_error_handler(
+            _child_gpid,
+            std::bind(&replica_split_manager::child_handle_split_error,
+                      std::placeholders::_1,
+                      "update_child_group_partition_count failed, "
+                      "wrong partition status or split status"));
+        _replica->_primary_states.sync_send_write_request = false;
+        parent_cleanup_split_context();
```

Comment:
Seems there are some similar code on error handling paths, could you do some refactor?

Reply:
Yes, split error handling has little difference in different error cases, I will refactor it after I merge core implementation.